tests/unit/lib/upload_manager_test.py

```diff
@@ -23,110 +29,113 @@ def update_bundle(self, *args, **kwargs):
-            f.write(tar_gzip_directory(source_dir).read())
-        self.do_upload([source], simplify_archives=True)
-        self.assertTrue(os.path.exists(source))
+    def test_fileobj_single(self):
+        self.do_upload([('source', BytesIO(b'testing'))])
         self.check_file_contains_string(self.bundle_location, 'testing')
 
-    def test_single_local_path_remove_sources(self):
-        source = os.path.join(self.temp_dir, 'filename')
-        self.write_string_to_file('testing', source)
-        self.do_upload([source], remove_sources=True)
-        self.assertFalse(os.path.exists(source))
-
-    def test_single_local_gzip_path_remove_sources(self):
-        source = os.path.join(self.temp_dir, 'filename.gz')
-        self.write_bytes_to_file(gzip_bytestring(b'testing'), source)
-        self.do_upload([source], remove_sources=True)
-        self.assertFalse(os.path.exists(source))
-
-    def test_single_fileobj(self):
-        self.do_upload([('source', BytesIO(b'testing'))])
+    def test_fileobj_single_gz(self):
+        self.do_upload([('source.gz', BytesIO(gzip_bytestring(b'testing')))])
         self.check_file_contains_string(self.bundle_location, 'testing')
 
-    def test_single_fileobj_tar_gz_simplify_archives(self):
+    def test_fileobj_single_tar_gz_simplify_archives(self):
         source = os.path.join(self.temp_dir, 'source_dir')
         os.mkdir(source)
         self.write_string_to_file('testing', os.path.join(source, 'filename'))
         self.do_upload([('source.tar.gz', tar_gzip_directory(source))])
         self.check_file_contains_string(self.bundle_location, 'testing')
 
-    def test_single_fileobj_tar_gz_no_simplify_archives(self):
+    def test_fileobj_single_tar_gz_no_simplify_archives(self):
         source = os.path.join(self.temp_dir, 'source_dir')
         os.mkdir(source)
         self.write_string_to_file('testing', os.path.join(source, 'filename'))
         self.do_upload([('source.tar.gz', tar_gzip_directory(source))], simplify_archives=False)
         self.assertEqual(['filename'], os.listdir(self.bundle_location))
         self.check_file_contains_string(os.path.join(self.bundle_location, 'filename'), 'testing')
 
+    def test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive(self):
+        """If the user included two files, README and .DS_Store, in the archive,
+        the archive should not be simplified because we have more than one file in the archive.
+        """
+        source = os.path.join(self.temp_dir, 'source_dir')
+        os.mkdir(source)
+        self.write_string_to_file('testing', os.path.join(source, 'README'))
+        self.write_string_to_file('testing', os.path.join(source, '.DS_Store'))
+        self.do_upload([('source.tar.gz', tar_gzip_directory(source))])
+        self.assertEqual(['.DS_Store', 'README'], sorted(os.listdir(self.bundle_location)))
+
+    def test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive_2(self):
```

Comment:
It has 3 files instead of 2 files. Before this PR, the archive would be simplified in test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive and not be simplified in test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive_2. With this PR, the archive would not be simplified in both cases.